books/free-programming-books-subjects.md

```diff
@@ -531,6 +531,10 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Encyclopedia of Human Computer Interaction 2nd Edition](https://www.interaction-design.org/literature/book/the-encyclopedia-of-human-computer-interaction-2nd-ed)
 * [Essential Image Optimization](https://images.guide) - Addy Osmani
 * [Foundations of Programming](https://openmymind.net/FoundationsOfProgramming.pdf) - Karl Seguin (PDF)
+* [Front-End Developer Handbook 2016](https://frontendmasters.com/guides/front-end-handbook/2016/) - Cody Lindley, Frontend Masters (HTML)
+* [Front-End Developer Handbook 2017](https://frontendmasters.com/guides/front-end-handbook/2017/) - Cody Lindley, Frontend Masters (HTML)
+* [Front-End Developer Handbook 2018](https://frontendmasters.com/guides/front-end-handbook/2018/) - Cody Lindley, Frontend Masters (HTML)
+* [Front-End Developer Handbook 2019](https://frontendmasters.com/guides/front-end-handbook/2019/) - Cody Lindley, Frontend Masters (HTML, [PDF](https://github.com/FrontendMasters/front-end-handbook-2019/raw/HEAD/exports/Front-end%20Developer%20Handbook%202019.pdf), [EPUB](https://github.com/FrontendMasters/front-end-handbook-2019/raw/HEAD/exports/Front-End%20Developer%20Handbook%202019.epub))
```

Comment:
Only one link is needed for this book. Please remove the format specific links.

Reply:
Resolved by commit 0f9aab2d717e35fee0cd0774e51e38f36f2b4eae